tests/BuildScriptGenerator.Tests/SemanticVersionResolverTest.cs

```diff
@@ -94,5 +95,40 @@ public void SpecificVersionSpecified()
             // Assert;
             Assert.Equal("1.2.3", returnedVersion);
         }
+
+        [Theory]
+        [InlineData("1.2", "1.2.3", -1)]
+        [InlineData("1.3.4", "1.2.3", 1)]
+        [InlineData("3.5", "2.1.1", 1)]
+        [InlineData("3.5", "4.1", -1)]
+        public void CompareVersion_ProvidedVersionToSupportedVersion(
+            string providedVersion, 
+            string supportedVersion, 
+            int resultExpected)
+        {
+            // Arrange and Act 
+            int comparisonResult= SemanticVersionResolver.CompareVersions(providedVersion, supportedVersion);
+
+            // Assert 
+            Assert.Equal(comparisonResult, resultExpected);
+        }
+
+        [Theory]
+        [InlineData("1", "1.2.3", int.MinValue)]
```

Comment:
Looking at code of `CompareVersions`, it appears that the value of `int.MinValue` would occur when the parsing of the version like let's say `1` here would throw an exception, right?
Do you think we should instead assume `1` to be `1.0.0` and then do a comparison to get the value?

Reply:
actually I wanted to test the exception, that's the reason for this test. Usually if the provided version is "1" then we convert it to max satisfying version for example 1.2.4 and then this method gets called.